Feature/ab#44 ui framework

vyruz1986 opened the pull request.

vyruz1986 committed:
feat(ClientApp): Removed bootstrap, added material, converted menu

This change removes bootstrap from the project, adds material design,
and converts the navigation menu to material

Closes AB#44

vyruz1986 committed:
feat(ClientApp): Finished bootstrap > material conversion

All UI pages are now using material instead of bootstrap

Closes AB#44

vyruz1986 committed:
feat(About): Added about page and API to show version info

This page will show the UI version (from package.json) and API version
(using System.Relfection.Assembly)

Closes AB#44

BerendWouters commented on ClientApp/src/app/material.module.ts:
```diff
@@ -0,0 +1,33 @@
+      MatTableModule,
+   ],
+})
+export class MaterialModule {}
```
Nice, this will allow us to easility introduce new Material components.

BerendWouters commented on ClientApp/src/app/components/about/about.component.ts:
```diff
@@ -0,0 +1,29 @@
+})
+export class AboutComponent implements OnInit {
+   ngOnInit(): void {
+      throw new Error('Method not implemented.');
```
If the ngOnInit is not required, just remove the OnInit implementation.

BerendWouters commented on ClientApp/src/environments/environment.prod.ts:
```diff
@@ -1,3 +1,4 @@
 export const environment = {
    production: true,
+   VERSION: require('../../package.json').version,
```
Some explanation on how to use version information might be useful at the next meeting, I'm curious.

BerendWouters commented on Controllers/AboutController.cs:
```diff
@@ -0,0 +1,29 @@
+         Assembly assembly = Assembly.Load("Singer");
+         var gitVersionInformationType = assembly.GetType("GitVersionInformation");
+         var fullSemVerField = gitVersionInformationType.GetField("FullSemVer");
+         _apiVersion = fullSemVerField.GetValue(null).ToString();
```
We might want to introduce a service for this, but for now it's OK.

BerendWouters commented on Controllers/AboutController.cs:
```diff
@@ -0,0 +1,29 @@
+      }
+      public AboutModel Index()
+      {
+         return new AboutModel
```
Return objects should be DTOs (Data Transfer objects), allowing to easily distinguish between model logic and pure data objects.

BerendWouters commented on ClientApp/src/styles.css:
```diff
@@ -1 +1,2 @@
 /* You can add global styles to this file, and also import other style files */
+@import "~@angular/material/prebuilt-themes/indigo-pink.css";
```
Pink, yea? :D

BerendWouters reviewed (changes requested):
~Note, I haven't checked out this code.~ 
**Edit:** Have checked the code, it is working, and looks good too!

BerendWouters committed:
refactor: introduce DTOs

AB#44

BerendWouters committed:
refactor: introduce DTOs

AB#44

BerendWouters committed:
Merge branch 'feature/AB#44-ui-framework' of https://github.com/Brixel/Singer into feature/AB#44-ui-framework

BerendWouters reviewed (changes requested):
I've introduced a DTO folder and and AboutDTO. Other issues still need to be addressed (like the `ngOnInit()` )

vyruz1986 committed:
fix(AboutComponent): Removed not implemented OnInit function

Closes AB#44

vyruz1986 commented:
> I've introduced a DTO folder and and AboutDTO. Other issues still need to be addressed (like the `ngOnInit()` )

I fixed the `OnInit()` issues, I don't think there were other issues besides this one?

BerendWouters commented:
Seems ok to me
